tests/func/test_checkout.py

```diff
@@ -880,3 +880,31 @@ def test_checkout_executable(tmp_dir, dvc):
         assert not isexec
     else:
         assert isexec
+
+
+def test_checkout_partial(tmp_dir, dvc):
+    tmp_dir.dvc_gen(
+        {"db": {"a.txt": "a", "b.txt": "b", "sub_dir": {"c.txt": "c"}}}
+    )
+
+    db_dir = tmp_dir / "db"
+    sub_dir = db_dir / "sub_dir"
+    file_1, file_2, sub_dir_file = [
+        db_dir / filename for filename in ("a.txt", "b.txt", sub_dir / "c.txt")
+    ]
+
+    shutil.rmtree(db_dir)
+
+    dvc.checkout(targets=[str(file_1)])
+    assert db_dir.exists()
+    assert file_1.exists()
+    assert not file_2.exists()
+    assert not sub_dir.exists()
+
+    dvc.checkout(targets=[str(sub_dir_file)])
+    assert sub_dir.exists()
+    assert sub_dir_file.exists()
+    assert not file_2.exists()
```

Comment:
Similarly as https://github.com/iterative/dvc/pull/5163/files#r549324917,
```suggestion
    assert db_dir.read_text() == {"a.txt": "a", "sub_dir": {"c.txt": "c"}}
```

Reply:
I wasn't aware of this functionality, thanks @skshetry for letting me know!